app/assets/javascripts/autocomplete.js.coffee

```diff
@@ -50,7 +50,7 @@
       target.val ""
     else if $t.data('matched_val') != typed
       mySource.remote.get typed, (matches) ->
-        if matches.length > 0
+        if matches.length == 1
```

Comment:
This works-around the problem of selecting the first match when there are >1 matches.

Reply:
:+1: 

Also, I see that upstream has changed the order `matches.length` and `!$validate`. I think your approach offers more advantages.

https://github.com/avalonmediasystem/avalon/blob/062f6e644089abee8711227be756fdcbade70fdd/app/assets/javascripts/autocomplete.js.coffee#L53-L56